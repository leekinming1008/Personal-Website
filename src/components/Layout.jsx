import React from 'react';
import PropTypes from 'prop-types';
import Introduction from './Introduction';

function Layout({ children }) {
  return (
    <>
      <header>
        <Introduction />
      </header>
      <main>{children}</main>
      <footer className="flex items-center justify-center p-4">
        <a
          className="text-secondary hover:text-yellow-600"
          href="https://github.com/LaythAlqadhi"
          target="_blank"
        >
          Layth Alqadhi Â© {new Date().getFullYear()}
        </a>
        <span className="text-secondary px-4">|</span>
        <a
          className="text-secondary hover:text-yellow-600"
          href="https://github.com/LaythAlqadhi/personal-website-client-side"
          target="_blank"
        >
          My portfolio is open source
        </a>
      </footer>
    </>
  );
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
