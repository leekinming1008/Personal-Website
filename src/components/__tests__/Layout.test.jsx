import React from 'react';
import { render, screen } from '@testing-library/react';
import Layout from '../Layout';

describe('Layout', () => {
  it('should render Layout component correctly', () => {
    render(
      <Layout>
        <div />
      </Layout>,
    );

    expect(screen.container).toMatchSnapshot();
  });
});
