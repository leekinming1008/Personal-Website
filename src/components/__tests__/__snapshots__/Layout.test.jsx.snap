// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > should render Layout component correctly 1`] = `undefined`;
